### PR TITLE
refactor webview

### PR DESCRIPTION
**Pull request by DmytroHryshyn**

### CODE REVIEW

File: src/components/webview/WebviewPanel.ts [hunk 1]
```diff
@@ -24,9 +24,34 @@ export abstract class IntuitaWebviewPanel {
...
 		protected readonly _messageBus: MessageBus,
-	) {
+	) {}
+
+	setTitle = (title: string) => {
+		if (this._panel === null) {
+			return;
+		}
+
+		this._panel.title = title;
```
Comment (grzpab) [inline src/components/webview/WebviewPanel.ts:36]: `this._panel?.title = title;`

File: src/components/webview/WebviewPanel.ts [hunk 2]
```diff
@@ -24,9 +24,34 @@ export abstract class IntuitaWebviewPanel {
...
+
+	protected _attachExtensionEventListeners?(): void;
+	protected _attachWebviewEventListeners?(): void;
+
+	protected _postMessage(message: WebviewMessage) {
+		if (this._panel === null) {
+			return;
+		}
+
+		this._panel.webview.postMessage(message);
```
Comment (grzpab) [inline src/components/webview/WebviewPanel.ts:47]: `this._panel?.webview.postMessage(message);`

File: src/components/webview/WebviewPanel.ts (hunk 1 above)
Comment (DmytroHryshyn) [reply src/components/webview/WebviewPanel.ts:36]: why do we need `?` if we have null check on line 32?

Comment (grzpab) [reply src/components/webview/WebviewPanel.ts:36]: @DmytroHryshyn it's just more concise (I care a lot about readability and being able to quickly scan a file with one's eyes)

Comment (grzpab) [reply src/components/webview/WebviewPanel.ts:36]: I just checked and what I wanted to do is not possible here.

**Review by grzpab — APPROVED** [review]

**Pull request merged**